python/cuml/model_selection/_split.py

```diff
@@ -480,3 +481,128 @@ def train_test_split(X,
         return X_train, X_test, y_train, y_test
     else:
         return X_train, X_test
+
+
+class StratifiedKFold:
+    """
+    A cudf based implementation of Stratified K-Folds cross-validator.
+
+    Provides train/test indices to split data into stratified K folds.
+    The percentage of samples for each class are maintained in each
+    fold.
+
+    Parameters
+    ----------
+    n_splits : int, default=5
+        Number of folds. Must be at least 2.
+    shuffle : boolean, default=False
+        Whether to shuffle each class's samples before splitting.
+    random_state : int (default=None)
+        Random seed
+
+    Examples
+    --------
+    Splitting X,y into stratified K folds
+    .. code-block:: python
+        import cupy
+        X = cupy.random.rand(12,10)
+        y = cupy.arange(12)%4
+        kf = StratifiedKFold(n_splits=3)
+        for tr,te in kf.split(X,y):
+            print(tr, te)
+    Output:
+    .. code-block:: python
+        [ 4  5  6  7  8  9 10 11] [0 1 2 3]
+        [ 0  1  2  3  8  9 10 11] [4 5 6 7]
+        [0 1 2 3 4 5 6 7] [ 8  9 10 11]
+
+    """
+
+    def __init__(self, n_splits=5, shuffle=False, random_state=None):
+        if n_splits < 2 or not isinstance(n_splits, int):
+            raise ValueError(
+                f'n_splits {n_splits} is not a integer at least 2')
+
+        if random_state is not None and not isinstance(random_state, int):
+            raise ValueError(f'random_state {random_state} is not an integer')
+
+        self.n_splits = n_splits
+        self.shuffle = shuffle
+        self.seed = random_state
+        self.tpb = 64  # threads per bloc
```

Comment:
Is this configurable or why do we save it as an attribute of the class?

Reply:
Good catch. I was hesitant to make it configurable or not. I think setting it to 64 would be just fine. I'll make the change.